modules/GKB/CommonUtils.pm

```diff
@@ -319,81 +379,76 @@ sub get_unique_species {
                 !XOR(($species_instance->displayName =~ /virus/) ? 1 : 0, ($seen_species_display_name =~ /virus/) ? 1 : 0)
             } @unique_species_instances);
     }
-    
-    return sort { $a->displayName cmp $b->displayName } @unique_species_instances;    
+
+    my @sorted_unique_species_instances = sort { $a->displayName cmp $b->displayName } @unique_species_instances;
+    return @sorted_unique_species_instances;
 }
 
 sub XOR {
     my $expression1 = shift;
     my $expression2 = shift;
-    
+
     return ($expression1 || $expression2) && (!($expression1 && $expression2));
 }
 
-sub get_components
-{
-	my $composite_entity = shift;
-
-	my @components;
-	if ($composite_entity)
-	{
-		foreach my $component (@{$composite_entity->hasComponent}, @{$composite_entity->hasMember},
-								@{$composite_entity->hasCandidate}, @{$composite_entity->repeatedUnit})
-		{
-			push @components, $component;
-			my @sub_components = grep {defined} get_components($component);
-			if (@sub_components)
-			{
-				push @components, @sub_components;
-			}
-		}
-	}
-
-	return @components;
+sub get_components {
+    my $composite_entity = shift;
+
+    my @components;
+    if ($composite_entity) {
+        foreach my $component (
```

Comment:
The format of this block takes a bit to understand. It might be easier if all of the arrays that will be iterated over were on the same line (instead of the two lines), or if all array elements were put into a single array before the for loop. This is obviously a style thing, I just felt it went against the code flow

Reply:
Yup, that's a good point.  I'll change it.